### PR TITLE
Hotfix: Rosetta include memos in block response metadata

### PR DESCRIPTION
**Pull request by bkase**
Explain your changes:
* Adds to_string_hum to memo and renames to_string/of_string appropriately
* Adds memo to block responses
* Removes unused transactions from Rosetta

Explain how you tested your changes:
* Tried against live data in the devnet

### CODE REVIEW

File: src/app/rosetta/lib/block.ml [hunk 1]
```diff
@@ -790,7 +790,18 @@ module Specific = struct
                       { Transaction.transaction_identifier=
                           {Transaction_identifier.hash= info.hash}
                       ; operations= User_command_info.to_operations' info
-                      ; metadata= None } )
+                      ; metadata= Option.bind info.memo ~f:(fun memo ->
```
Comment (psteckler) [inline src/app/rosetta/lib/block.ml:793]: I would use `base58` or `base58_check` for `memo`

and `memo` for `memo'`

File: src/app/rosetta/lib/block.ml [hunk 2]
```diff
@@ -790,7 +790,18 @@ module Specific = struct
...
+                        try
+                          let memo' =
+                            let open Mina_base.Signed_command_memo in
+                            memo |> of_base58_check_exn |> to_string_hum
+                          in
+                          if String.equal memo' "" then
```
Comment (psteckler) [inline src/app/rosetta/lib/block.ml:799]: `String.is_empty`

File: src/lib/mina_base/signed_command_memo.mli [hunk 3]
```diff
@@ -44,9 +44,14 @@ val dummy : t
...
-val of_string : string -> t
+val of_base58_check_exn : string -> t
+
+(** for a memo of bytes, return a plaintext string
+    for a memo of a digest, return a hex-encoded string, prefixed by '0x'
+*)
+val to_string_hum : t -> string
 
 (** is the memo a digest *)
 val is_digest : t -> bool
```
Comment (psteckler) [inline src/lib/mina_base/signed_command_memo.mli:57]: should probably expose `is_bytes` here

**Review by psteckler — APPROVED** [review]

**Review by MartinMinkov — APPROVED** [review]

**Comment by lk86** [comment]
This will fix the docker image pushes https://github.com/MinaProtocol/mina/pull/10344/files

**Review by lk86 — APPROVED** [review]

**Comment by bkase** [comment]
!approved-for-mainnet

**Pull request merged**